include/libxnvme_kvs.h

```diff
@@ -39,7 +40,7 @@ extern "C" {
  */
 int
 xnvme_kvs_retrieve(struct xnvme_cmd_ctx *ctx, uint32_t nsid, const void *key, uint8_t key_len,
-		   uint8_t opt, const void *dbuf, size_t dbuf_nbytes);
+		   const void *dbuf, uint32_t dbuf_nbytes, uint8_t opt);
```

Comment:
> 1. The original intend was to 'fixup' (i.e. squash without commit message) the commit on the first KV commit.
> 

I see. Since the PR is a continuation of Eduard's work, then I would prefer that his original work is kept as-is and then, that the changes to that work are described by commits "on top".  There is a bit of history-fiddling to achieve that, e.g. either split the commit currently named "Squash this with first commit " into individual commits, for the changes in that commit which looks like it at least does:

* Change license from APACHE-2 to BSD-3-Clause
* Wrap definitions in enum
* Change signature of functions "x, y, z" dbuf -> vuf
* Change signature: move opt to end
* Re-factor ``kvs_put_key_into_cmd`` to ``kvs_cmd_set_key``

The most helpful part of making such as split is that one can read the motivations behind the changes.

In case  you do not want to spend time making this split, then just let me know, I will then do it and force-push the changes to your branch, for doing so, let me know when I can push without stomping on your branch.

>     2. This can be changed
> 

Great, thanks!

>     3. It was either because we wanted to keep the order of the other arguments the same across different API-function signatures, or we found an example that also had `opts` at the end.

I see. Thanks for clarifying. It would be great if the motivation was part of the commit as noted above.

Reply:
You're very welcome to do the split. Then you'll also get it how you want it.

Reply:
Ok, done. Without alterations to changes in the commits, just a history-rewrite splitting the commit ``Squash this with first commit`` into the following commits:

* xnvmec/kvs: add missing ','
* tests/kvs: adjust to changes in xnvmec-options
* api/xnvmec: add XNVMEC_OPT_KV_STORE_COMPRESS and adjust format
* kvs: rename helper from kvs_put_key_into_cmd() to kvs_cmd_set_key()
* api/kvs,tests/kvs,tools/kvs: change license to BSD-3-Clause
* api/kvs: move KVS-related definitions from spec to kvs
* api/kvs: change signature of xnvme_kvs_{retrieve,store,list}